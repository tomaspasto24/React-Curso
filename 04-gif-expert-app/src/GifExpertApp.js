import React, { useState } from 'react';
import { AddCategory } from './Components/AddCategory';
import { GifGrid } from './Components/GifGrid';

const GifExpertApp = ( { defaultCategories = [] } ) => {
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />

            <ol>
                { categories.map( cat =>
                    <GifGrid 
                    key={cat}
                    category={cat} 
                    />
                )}
            </ol>
        </>
    )
};

export default GifExpertApp;