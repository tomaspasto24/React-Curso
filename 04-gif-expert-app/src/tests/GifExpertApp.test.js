import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas sobre GifExpertApp.', () => {
    
    test('Que muestre correctamente el componente.', () => {

        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch', 'Mario'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    });

});