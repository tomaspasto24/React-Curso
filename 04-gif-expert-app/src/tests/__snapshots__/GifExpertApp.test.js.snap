// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExpertApp. Que muestre correctamente el componente. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
